File: booking-service/src/main/java/com/study/booking/application/dto/BookingResponse.java
package com.study.booking.application.dto;

import com.study.booking.domain.entity.Booking;
import com.study.booking.domain.entity.BookingStatus;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class BookingResponse {
    private Long id;
    private Long userId;
    private Long tripId;
    private BookingStatus status;
    private LocalDateTime bookedAt;
    private LocalDateTime createdAt;

    public static BookingResponse from(Booking booking) {
        return BookingResponse.builder()
                .id(booking.getId())
                .userId(booking.getUserId())
                .tripId(booking.getTripId())
                .status(booking.getStatus())
                .bookedAt(booking.getBookedAt())
                .createdAt(booking.getCreatedAt())
                .build();
    }
}

File: booking-service/src/main/kotlin/com/study/booking/application/dto/BookingResponse.kt
package com.study.booking.application.dto

import com.study.booking.domain.entity.Booking
import com.study.booking.domain.entity.BookingStatus
import java.time.LocalDateTime

data class BookingResponse(
    val id: Long, val userId: Long, val tripId: Long,
    val status: BookingStatus, val bookedAt: LocalDateTime, val createdAt: LocalDateTime
) {
    companion object {
        fun from(booking: Booking) = BookingResponse(
            id = booking.id, userId = booking.userId, tripId = booking.tripId,
            status = booking.status, bookedAt = booking.bookedAt, createdAt = booking.createdAt
        )
    }
}
